src/misc/libraries/BytesLib.sol

```diff
@@ -6,12 +6,25 @@ pragma solidity 0.8.28;
 ///         The library lets you concatenate, slice and type cast bytes arrays both in memory and storage.
 /// @author Modified from Solidity Bytes Arrays Utils v0.8.0
 library BytesLib {
+    error SliceOverflow();
+    error SliceOutOfBounds();
+    error SliceToAddressOutOfBounds();
+    error SliceToUint8OutOfBounds();
+    error SliceToUint16OutOfBounds();
+    error SliceToUint32OutOfBounds();
+    error SliceToUint64OutOfBounds();
+    error SliceToUint128OutOfBounds();
+    error SliceToUint256OutOfBounds();
+    error SliceToBytes32OutOfBounds();
+    error SliceToBytes16OutOfBounds();
+    error SliceToBoolOutOfBounds();
```

Comment:
Should we just add `SliceOutOfBounds()`?

Reply:
I think that is a reasonable simplification yeah

Reply:
Will do then

Reply:
Done